src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/tools/JMap.java

```diff
@@ -50,16 +50,21 @@ public String getName() {
     }
 
     protected String getCommandFlags() {
-        return "-heap|-heap:format=b|-histo|-clstats|-finalizerinfo";
+        return "-heap|-heap:format=b[,gz=<1-9>]|-histo|-clstats|-finalizerinfo";
     }
 
     protected void printFlagsUsage() {
-        System.out.println("    <no option>\tto print same info as Solaris pmap");
-        System.out.println("    -heap\tto print java heap summary");
-        System.out.println("    -heap:format=b\tto dump java heap in hprof binary format");
-        System.out.println("    -histo\tto print histogram of java object heap");
-        System.out.println("    -clstats\tto print class loader statistics");
-        System.out.println("    -finalizerinfo\tto print information on objects awaiting finalization");
+        System.out.println("    <no option>\tTo print same info as Solaris pmap.");
+        System.out.println("    -heap\tTo print java heap summary.");
+        System.out.println("    -heap:format=b[,gz=<1-9>]\tTo dump java heap in hprof binary format.");
+        System.out.println("                             \tIf gz specified, the heap dump is written");
+        System.out.println("                             \tin gzipped format using the given compression level.");
+        System.err.println("                             \t1 (recommended) is the fastest, 9 the strongest compression.");
+        System.out.println("    -heap:format=x           \tTo dump java heap in GXL format.");
+        System.out.println("                             \tPlease be aware that \"gz\" option is not valid for heap dump in GXL format.");
```

Comment:
I just noticed that there is a (pre-existing) `file=<dumpfile>` argument that can be used with `-heap:format=` to specify the dumpfile name. Otherwise it dumps to `heap.bin`. It's not currently documented. I think it's enough to just add `[,file=<dumpfile>]` to the argument syntax. No description should be necessary.